docs/contributor/developing.md

```diff
@@ -73,21 +73,30 @@
 
 ## Testing
 
-1. Run a Basic Syntax/Lint Check (using flake8) and resolve any issues
+PIP modules used in this section such as `ruff` and `pydocstyle` are installed when you run `pip install -e .[dev]`
+as described above.
 
-    ``` cmd
-    flake8 .
-    ```
+> See [`pyproject.toml`](../../pyproject.toml) for the full list of development dependencies.
+
+1. Run a Basic Syntax/Lint Check (using ruff) and resolve any issues
+
+    * Run ruff
+
+      ``` cmd
+      ruff check .
+      ```
 
-    INFO: Newer editors are very helpful in resolving source formatting errors
-    (whitespace, indentation, etc). In VSCode open the py file and use
-    ++alt+shift+f++ to auto format.
+      * Note: Newer editors are very helpful in resolving source formatting errors. For example, in VSCode, you can
+              open the Python file and use `Alt+Shift+F` to auto format. See the [Ruff VSCode extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
+              for more information.
 
 2. Run a Basic Python docstring Check (using pydocstring) and resolve any issues
 
-    ``` cmd
-    pydocstyle edk2toollib
-    ```
+    * Run pydocstyle
```

Comment:
`ruff` is used for the `pydocstyle` formatting now. Rule set "E" is for pydocstyle errors and rule set "W" is for pydocstyle errors.

Reply:
Updated with those details.